sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
+
+    The check is performed by calculating the distance of every pixel
+    from the center of the Sun.  If they are all on-disk, then the
+    function returns True. Otherwise, the function returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    is_all_on_disk : `~bool`
+        Returns True if all map pixels are strictly less than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def contains_limb(smap):
+    """
+    Checks if a map contains a portion of the solar limb.
+
+    The check is performed by calculating the distance of every pixel from
+    the center of the Sun.  If at least one pixel is on disk and at least one
+    pixel is off disk, the function returns True.  Otherwise, the function
+    returns False. Note that this function will also true if the entire solar
+    limb is within the field of view of the map.  Note also that in the case
+    of coronagraph images the limb itself need not be observed.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    contains_limb : `~bool`
+        Returns True If at least one pixel is on disk and at least one pixel
+        is off disk.
+    """
+    on_disk = coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs)
+    return np.logical_and(np.any(on_disk), np.any(~on_disk))
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's distance
+    from the center of the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate.  The frame property of the input
+        coordinate must be helioprojective.
+
+    scale : `~astropy.units.Quantity`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    is_on_disk : `~bool`
+        Returns True if the coordinate is on disk, False otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    if not isinstance(coordinate.frame, Helioprojective):
+        raise ValueError("Input coordinate must be in the Helioprojective Cartesian frame.")
+    return u.R_sun * (np.sqrt(coordinate.Tx ** 2 + coordinate.Ty ** 2) / scale) < 1 * u.R_sun
+
+
+def on_disk_bounding_coordinates(smap):
+    """
+    Returns the the bottom left and top-right coordinates of a minimal
+    rectangular region that contains all the on-disk pixels in the input map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    (bl, tr) : `~list`
```

Comment:
I would think the best way to do this would be to return one SkyCoord, but if not I think this should read:

```
bl, tr : `~astropy.coordinates.SkyCoord`
```

Reply:
done